cradle-core/src/test/java/com/exactpro/cradle/BookInfoTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2024 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.cradle;
+
+import org.testng.annotations.Test;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertSame;
+
+public class BookInfoTest {
+    public static Random RANDOM = new Random();
+    public static final BookId BOOK_ID = new BookId("test-book");
+    public static final List<PageInfo> PAGES;
```

Comment:
This is of course just a test, however, why do we have a public mutable static field? This either should be private or (the better option in my opinion) be immutable. You can use `List.copyOf` to make an immutable collection after you collect all the pages into the ArrayList